### ISSUE TITLE
`heading-capitalization` (`MD063`) - false positive when heading contains a link

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#844]
### rumdl version

0.2.62

### Installation method

Other

### Bug description

when a heading contains a link and `style` is set to `sentence-case`, it incorrectly suggests removing the capital at the start

### Steps to reproduce

```md
# [Foo bar](https://example.com)
```

### Expected behavior

no error

### Actual behavior

```
README.md:1:3: [MD063] Heading should use sentence case: '[Foo bar](https://example.com)' -> '[foo bar](https://example.com)' [*]
```

### Configuration

```toml
[global]
enable = ['ALL']
[MD063]
style = 'sentence-case'
```

### Operating system

Windows

**Comment by rvben** [comment]
Thanks @DetachHead. The leading `[` made `Foo` look like a mid-sentence word rather than the first word of the heading, so sentence case lowercased it.

`# [Foo bar](https://example.com)` is clean as of v0.2.64.
